### PR TITLE
Split uptodate state

### PR DESCRIPTION
**Pull request by parlt91**
## What does this PR change?

Split the `uptodate` state into `uptodate`, `reboot` and `rebootifneeded`.
Force reboot state to be last in order for recurring states.
Don't allow `rebootifneeded` and `reboot` to be used together.

## GUI diff

No difference.

- [x] **DONE**

## Documentation
- [Documentation PR](https://github.com/uyuni-project/uyuni-docs/pull/3184)

- [x] **DONE**

## Test coverage

- No tests: 

- [x] **DONE**

## Links

Issue(s): https://github.com/SUSE/spacewalk/issues/22483

- [x] **DONE**

## Changelogs

Make sure the changelogs entries you are adding are compliant with https://github.com/uyuni-project/uyuni/wiki/Contributing#changelogs and https://github.com/uyuni-project/uyuni/wiki/Contributing#uyuni-projectuyuni-repository

If you don't need a changelog check, please mark this checkbox:

- [ ] No changelog needed

If you uncheck the checkbox after the PR is created, you will need to re-run `changelog_test` (see below)

## Re-run a test

If you need to re-run a test, please mark the related checkbox, it will be unchecked automatically once it has re-run:

- [ ] Re-run test "changelog_test"
- [ ] Re-run test "backend_unittests_pgsql"
- [ ] Re-run test "java_pgsql_tests"
- [ ] Re-run test "schema_migration_test_pgsql"
- [ ] Re-run test "susemanager_unittests"
- [ ] Re-run test "javascript_lint"
- [ ] Re-run test "spacecmd_unittests"

# Before you merge

Check [How to branch and merge properly](https://github.com/uyuni-project/uyuni/wiki/How-to-branch-and-merge-properly)!

### CODE REVIEW

**Review by cbbayburt — APPROVED** [review]
LGTM :+1: (I haven't tried the new states though)

File: web/html/src/components/states-picker.tsx [hunk 1]
```diff
@@ -72,6 +72,7 @@ class StatesPicker extends React.Component<StatesPickerProps, StatesPickerState>
 
   constructor(props: StatesPickerProps) {
     super(props);
+    this.state.messages = [];
```
Comment (Etheryte) [inline web/html/src/components/states-picker.tsx:75]: You can move this array initialization to the `StatesPickerState` definition:

```tsx
class StatesPickerState {
  ...
  // TODO: Can we drop `any` here?
  messages: MessageType[] | any = [];
}
```

**Review by Etheryte — DISMISSED** [review]
Looks good. 👍

**Comment by deneb-alpha** [comment]
> @admd @deneb-alpha We were discussing during our standup if we could/should include this for GA. It's not a bug fix, but rather some small quality of life addition for users. Do you think we can put it in or should we wait for 1. MU?

From RC we are in feature freeze and unless I understood this GH issue wrong, the split is not fixing any bug or broken feature. Given that, I would personally prefer to see these changes in the first MU, but it's up to @admd to decide.

**Comment by admd** [comment]
> > @admd @deneb-alpha We were discussing during our standup if we could/should include this for GA. It's not a bug fix, but rather some small quality of life addition for users. Do you think we can put it in or should we wait for 1. MU?
> 
> From RC we are in feature freeze and unless I understood this GH issue wrong, the split is not fixing any bug or broken feature. Given that, I would personally prefer to see these changes in the first MU, but it's up to @admd to decide.

I agree with Marina. This can wait for 1st MU.

**Review by Etheryte — APPROVED** [review]

**Review by cbbayburt — APPROVED** [review]

**Pull request merged**